### PR TITLE
More Plumbing Fixes & Pill Press UI Changes

### LINKED ISSUE #79051: Chemical factories are still broken... Somewhat
Chemical factories appear to be fixed for the most part thanks to #78947 , but not the presses. Presses still lock up specifically when set to 50 unit bottle outputs, as can be seen here: 
QggilbX

I'm not sure entirely what happens on the code end, but it will produce one 50 unit bottle at first, and then refuse to do so again regardless with whatever you do to change it around and such.

@SyncIt21 All yours pal!

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request
1. Fixes #79051
It's a consequence of floating-point math. We round it to 2 decimal places to display results like 50 but in fact its actual value is something like 49.999... something. So, we also shift up our expectations and call it a day.
2. Made a lot of variables defines and lists static to save memory for plumbing pill press and moved global lists to it's global list folder
3. Copied over chem master patch & pill designs over too plumbing press and removed the old designs & resized the UI

## Changelog
:cl:
fix: plumbing pill press & bottler won't stop when processing 50 unit bottles
code: made a lot of variables defines and lists static to save memory for plumbing pill press. Moved global lists to it's rightful 
place
code: copied over chem master pill & patch designs over to plumbing pill press and removed the old designs. resized UI
/:cl:

### CODE REVIEW

**Comment by MTandi** [comment]
Please note that I have refactored the styles for the ChemMaster, and we proobably should reuse it for plumbing too. It's a lot simpler now and you only need to list the containers.

**Comment by SyncIt21** [comment]
> Please note that I have refactored the styles for the ChemMaster, and we proobably should reuse it for plumbing too. It's a lot simpler now and you only need to list the containers.

Do you mean the patch & pill styles? Yeah i tried to make them into global lists but the assets runtime when you attempt to load them during global list initialization.

**Comment by MTandi** [comment]
> Do you mean the patch & pill styles? Yeah i tried to make them into global lists but the assets runtime when you attempt to load them during global list initialization.

Please check `chem_master.dm` it already has `chem_master_containers` global list defined. Might want to move it elsewhere for proper initialization order.

**Comment by SyncIt21** [comment]
Ok i know what needs to be done but it's too big for the scope of this PR. Just 1 question though 

Since we need to use the sprites inside of `/datum/asset/spritesheet/chemmaster` for plumbing as well does that mean i now need to deprecate `/datum/asset/spritesheet/simple/pills` & `/datum/asset/spritesheet/simple/patches`? or are these sprite sheets used elsewhere?

**Comment by MTandi** [comment]
> Ok i know what needs to be done now but it's too big for the scope of this PR. Just 1 question though
> 
> Since we need to use the sprites inside of `/datum/asset/spritesheet/chemmaster` for plumbing as well does that mean i now need to deprecate `/datum/asset/spritesheet/simple/pills` and `/datum/asset/spritesheet/simple/patches` sprite sheets? or are these sprites used elsewhere?

They were used by chemmaster too, but now they are needed only for plubming and can be deprecated if you copy the logic from the chemmaster.

Sure, I just see that you touched the pill press and noted that we have a tech debt there.

**Comment by SyncIt21** [comment]
> They were used by chemmaster too, but now they are needed only for plubming and can be deprecated if you copy the logic from the chemmaster.
> 
> Sure, I just see that you touched the pill press and noted that we have a tech debt there.

Thanks for the idea by the way. I have made way too many PR's labelled no GBP. I can start with this :D

**Comment by SyncIt21** [comment]
So i got bored & decided to follow up on your request anyway. Both chem master & plumbing pill press now share the same pill & patch styles. There goes my GBP

File: code/modules/plumbing/plumbers/pill_press.dm [hunk 1]
```diff
@@ -147,12 +163,22 @@
...
-				max_volume = max_patch_volume
+				max_volume = MAX_PATCH_VOLUME
 			else if (product == "bottle")
-				max_volume = max_bottle_volume
-			current_volume = clamp(current_volume, min_volume, max_volume)
+				max_volume = MAX_BOTTLE_VOLUME
+			else
+				return
+			current_volume = clamp(current_volume, MIN_VOLUME, max_volume)
 		if("change_patch_style")
```
Comment (MTandi) [inline code/modules/plumbing/plumbers/pill_press.dm:176]: Does UI really need all these actions? In chem master I just select the container by ref, without caring whether it's pill, patch or a bottle.

```
	if(action == "selectContainer")
		selected_container = params["ref"]
		return TRUE
```

And then you can get the volume as `initial(container.volume)`

Comment (SyncIt21) [reply code/modules/plumbing/plumbers/pill_press.dm:176]: the constants `MAX_PILL_VOLUME` & `MAX_PATCH_VOLUME` are required for the UI(which is why i send them as static data) to know how to clamp user input. As for using ref yup I'll look into it

Comment (SyncIt21) [reply code/modules/plumbing/plumbers/pill_press.dm]: beautiful

**Comment by MTandi** [comment]
This shouldn't have [NO GBP] by now...

**Comment by SyncIt21** [comment]
> This shouldn't have [NO GBP] by now...

Fingers crossed for any maintainer gracious enough to remove the label. This took me a whole day to clean up

File: code/modules/plumbing/plumbers/pill_press.dm [hunk 2]
```diff
@@ -1,111 +1,116 @@
...
+	_decode_category()
+
 	AddComponent(/datum/component/plumbing/simple_demand, bolt, layer)
 
+/obj/machinery/plumbing/pill_press/examine(mob/user)
+	. = ..()
+	. += span_notice("The [name] currently has [stored_products.len] stored. There needs to be less than [MAX_FLOOR_PRODUCTS] on the floor to continue dispensing.")
+
+/// decode product category from it's type path and returns the decoded typepath
+/obj/machinery/plumbing/pill_press/proc/_decode_category()
```
Comment (Ghommie) [inline code/modules/plumbing/plumbers/pill_press.dm:66]:
```suggestion
/obj/machinery/plumbing/pill_press/proc/decode_category()
```

File: code/modules/plumbing/plumbers/pill_press.dm [hunk 3]
```diff
@@ -1,111 +1,116 @@
...
+	if(reagents.total_volume + (CHEMICAL_QUANTISATION_LEVEL * 10) >= current_volume)
+		var/obj/item/reagent_containers/container = locate(packaging_type)
+		container = new container(src)
+		var/suffix
+		if(packaging_category == CAT_PILLS)
+			suffix = "Pill"
+		else if(packaging_category == CAT_PATCHES)
+			suffix = "Patch"
+		else
+			suffix = "Bottle"
```
Comment (Ghommie) [inline code/modules/plumbing/plumbers/pill_press.dm:90]:
```suggestion
		switch(packaging_category)
			if(CAT_PILLS)
				suffix = "Pill"
			if(CAT_PATCHES)
				suffix = "Patch"
			else
				suffix = "Bottle"
```

File: code/modules/plumbing/plumbers/pill_press.dm [hunk 4]
```diff
@@ -1,111 +1,116 @@
...
+		container.name = "[product_name] [suffix]"
+		reagents.trans_to(container, current_volume)
+		stored_products += container
+
+	//dispense stored products on the floor
 	if(stored_products.len)
 		var/pill_amount = 0
 		for(var/thing in loc)
-			if(!istype(thing, /obj/item/reagent_containers/cup/bottle) && !istype(thing, /obj/item/reagent_containers/pill))
+			if(!istype(thing, /obj/item/reagent_containers))
```
Comment (Ghommie) [inline code/modules/plumbing/plumbers/pill_press.dm:99]: Github won't let me place a suggestion here, but you should just make the for loop typed: `for(var/obj/item/reagent_containers/thing)`

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/plumbing/plumbers/pill_press.dm (hunk 2 above)
Comment (SyncIt21) [reply code/modules/plumbing/plumbers/pill_press.dm:66]: I saw somewhere that if you want to create an internal private proc which is not meant to be used anywhere outside then you start it's name with underscore.

For example _SendSignal() that's what I was going for. But yeah I guess that's not exactly needed here

**Comment by SyncIt21** [comment]
@Ghommie it would be nice if you could remove the no gbp label.
I initially wanted to fix an bug I made with plumbing pill press in this [commit](https://github.com/tgstation/tgstation/pull/79059/commits/c933c2419601a517a10b0a643bdf7001928a7231)  but then I refactored it's UI snd everything as a whole afterwards so it would be nice if I could gain some points for it 😅?

**Pull request merged**